servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/MessageCodec.java

```diff
@@ -19,10 +19,21 @@
 import io.servicetalk.buffer.api.BufferAllocator;
 
 /**
- * Codec used to encode and decode gRPC messages.
- * This instance is shared across all requests/responses therefore it must provide thread safety semantics.
+ * Codec used to encode and decode {@link Buffer} content.
+ * This instance is shared therefore it must provide thread safety semantics.
  */
-public interface GrpcMessageCodec {
+public interface MessageCodec {
+
+    /**
+     * Take a {@link Buffer} and encode its contents resulting in a {@link Buffer} with the encoded contents.
+     *
+     * @param src the {@link Buffer} to encode
+     * @param allocator the {@link BufferAllocator} to use for allocating auxiliary buffers or the returned buffer
+     * @return {@link Buffer} the result buffer with the content encoded
+     */
+    default Buffer encode(Buffer src, BufferAllocator allocator) {
```

Comment:
Similar comment as I had for `ContentCodec`: it looks like only `encode(src, allocator) variant is enough for now. Can we remove an overload with offset and length? Users can slice the required `Buffer` portion in necessary to encode only a part of the `Buffer`.

Reply:
It is useful for gRPC (https://github.com/apple/servicetalk/pull/1198/files#diff-04a488534a209852ed234ffe65b56d079a09bae65dede9c6366116204cf00320R162) otherwise we have to create a slice or view. IMO this is cheaper

Reply:
IIUC, `lengthOfData` is always equal to `toDeserialize.readableBytes()`. Therefore, we do not need a slice.